src/gennylib/test/PhaseLoop_test.cpp

```diff
@@ -377,4 +378,29 @@ TEST_CASE("Actual Actor Example") {
             Catch::Matches("'Nop' cannot be used with any other keywords except 'Phase'. Check YML "
                            "configuration."));
     }
+
+    SECTION("SleepBefore and SleepAfter") {
+        using namespace std::literals::chrono_literals;
+        YAML::Node config = YAML::Load(R"(
+            SchemaVersion: 2018-07-01
+            Actors:
+            - Type: Inc
+              Phases:
+              - Repeat: 3
+                SleepBefore: 50 milliseconds
+                SleepAfter: 100 milliseconds
+                Key: 71
+        )");
+
+        auto imvProducer = std::make_shared<CounterProducer<IncrementsMapValues>>("Inc");
+        ActorHelper ah(config, 1, {{"Inc", imvProducer}});
+
+        auto start = std::chrono::high_resolution_clock::now();
+        ah.run();
+
+        auto duration = std::chrono::high_resolution_clock::now() - start;
```

Comment:
1. you've done the work to put the test clock into a new header is it worth using that to do the sleep? Like incorporate a `sleep` method into the clock?
2. depending on ↑, consider templating `Sleeper` on the clock similar to what the metrics types do. I.e. rename this to `SleeperT` and have a `using Sleeper = SleeperT<default-clock-or-whatever-foo>` line.